dynawo/sources/Solvers/SolverSIM/test/TestBasics.cpp

```diff
@@ -507,13 +507,13 @@ TEST(SimulationTest, testSolverSIMAlgebraicMode) {
 
   // Here we detect the algebraic mode change that occurs at t=2.
   solver->solve(tStop, tCurrent, y, yp, algebraicMode, discreteVariableChangeFound);
-  ASSERT_DOUBLE_EQUALS_DYNAWO(algebraicMode, true);
-  ASSERT_DOUBLE_EQUALS_DYNAWO(discreteVariableChangeFound, true);
+  ASSERT_EQ(algebraicMode, true);
+  ASSERT_EQ(discreteVariableChangeFound, true);
   ASSERT_DOUBLE_EQUALS_DYNAWO(y[2], 0.94766640118361411549);
   ASSERT_DOUBLE_EQUALS_DYNAWO(y[3], -0.09225375878818535547);
   ASSERT_DOUBLE_EQUALS_DYNAWO(y[10], 0);
   ASSERT_DOUBLE_EQUALS_DYNAWO(y[11], 1);
-  ASSERT_DOUBLE_EQUALS_DYNAWO(model->modeChangeAlg(), true);
+  ASSERT_EQ(model->getModeChangeType(), modeChangeType_t::ALGEBRAIC_J_UPDATE_MODE);
```

Comment:
Maybe we could also test getModeChange (the boolean method)?

Reply:
We can, but at the end of the solve() call, modeChange() will always return false. To properly test it, you should go lower in the solver calls and monitor it during the evalZMode loop.

Reply:
OK, not needed then